src/Microsoft.ML.Core/Utilities/DoubleParser.cs

```diff
@@ -17,6 +17,13 @@ internal static class DoubleParser
         private const ulong TopThreeBits = 0xE000000000000000UL;
         private const char InfinitySymbol = '\u221E';
 
+        // The decimal marker that separates the integer part from the fractional part of a number
+        // written in decimal from can vary across different cultures as either '.' or ','. The
+        // default decimal marker in ML .NET is '.', however through this static char variable,
+        // we allow users to specify the decimal marker used in their datasets as ',' as well.
+        [BestFriend]
+        internal static char DecimalMarker = '.';
+
```

Comment:
This static variable `DecimalMarker` is set to `'.'` by default, and can be changed to `','` in `TextLoader.BoundLoader`. As `DoubleParser` is a static class, there can only be one instance of this at one point, which allows for multiple datasets that have `','` as their decimal marker to be read in correctly. The alternative to this approach is overloading multiple functions in `TextLoaderParser`, `Conversion.cs` and `DoubleParser` just to add an additional parameter for `decimalMarker`, which would not be a good idea for backwards compatibility when new additions are added as well.

Reply:
Can you please explore the other option as well? Will we be actually breaking backward compatibility? 

---
In reply to: [428429053](https://github.com/dotnet/machinelearning/pull/5145#discussion_r428429053) [](ancestors = 428429053)

Reply:
I don't think that exploring this other option would break backward compatibility either. But I would expect the other solution to be somewhat more difficult to implement than the current solution on your PR.

Since the parse methods are supposed to be called as the _conv delegate through the ColumnPipes on the TextLoader, I _think_ (but I am not sure) that you won't be able to add another parameter to  `TryParse(in TX src, out R4 dst)` (since the _conv delegate would expect only 2 parameters on its signature)... so the solution I'd suggest would involve creating another method like `TryParseWithCommaDecimal(in TX, src out R4 dst)` and then change anything necessary in TextLoader and in DoubleParser so that the correct delegate is assigned to the ColumnPipe's _conv according to the TextLoader option.

---
In reply to: [428442264](https://github.com/dotnet/machinelearning/pull/5145#discussion_r428442264) [](ancestors = 428442264,428429053)